### PR TITLE
🏏 Conftest and Bats agent  for Jenkins 🦇

### PR DESCRIPTION
**Pull request by springdo**
#### What is this PR About?
Adding tools for running conf tests in a Jenkins pipeline. Inspired by @garethahealy 's GH Action. 

#### How do we test this?
build and run the image. 
image

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: jenkins-slaves/jenkins-slave-conftest/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM quay.io/openshift/origin-jenkins-agent-base:4.4
+
+ENV BATS_VERSION=1.2.0 \
+    YQ_VERSION=3.3.2 \
+    CONFTEST_VERSION=0.18.2
+
+RUN curl --fail -skL https://github.com/open-policy-agent/conftest/releases/download/v${CONFTEST_VERSION}/conftest_${CONFTEST_VERSION}_Linux_x86_64.tar.gz -o /tmp/conftest.tar.gz && \ 
+    tar -C /tmp -xzf /tmp/conftest.tar.gz && \
+    mv /tmp/conftest /usr/local/bin && \
+    curl --fail -sL  https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64 -o /usr/local/bin/yq && \
```
Comment (garethahealy) [inline jenkins-slaves/jenkins-slave-conftest/Dockerfile:10]: FYI: the yq used by the GH-Action isn't this ya, its: https://pypi.org/project/yq/ - which also requires JQ

File: jenkins-slaves/jenkins-slave-conftest/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Jenkins Reqo Slave 🦇
```
Comment (garethahealy) [inline jenkins-slaves/jenkins-slave-conftest/README.md:1]: Reqo? typo?

File: jenkins-slaves/jenkins-slave-conftest/Dockerfile (hunk 1 above)
Comment (springdo) [reply jenkins-slaves/jenkins-slave-conftest/Dockerfile:10]: JQ is pre installed on the slave from memory, but cheers for the heads up @garethahealy. I guess if your policies are using this python `yq` library i should update this pr to match

File: jenkins-slaves/jenkins-slave-conftest/README.md (hunk 2 above)
Comment (springdo) [reply jenkins-slaves/jenkins-slave-conftest/README.md:1]: lolz - I initially called it a rego slave but ovs find and replace didn't catch this one

**Comment by springdo** [comment]
bump @garethahealy

File: jenkins-slaves/jenkins-slave-conftest/Dockerfile [hunk 3]
```diff
@@ -1,18 +1,21 @@
-FROM quay.io/openshift/origin-jenkins-agent-base:4.4
+# needed to install the python yq library
```
Comment (garethahealy) [inline jenkins-slaves/jenkins-slave-conftest/Dockerfile:1]: put this after the FROM - i've seen docker/podman crash out if the first line isn't a FROM

Comment (springdo) [reply jenkins-slaves/jenkins-slave-conftest/Dockerfile]: `done();`

**Comment by springdo** [comment]
@garethahealy not sure why CI failed here, looks like a rando hangup trying to get my git repo which is odd.

/test cq-jenkins-slaves-pr-build

**Comment by garethahealy** [comment]
another flake

/test cq-jenkins-slaves-pr-build

**Pull request merged**